scipy/stats/_distribution_infrastructure.py

```diff
@@ -2361,6 +2444,43 @@ def _pdf_formula(self, x, **params):
     def _pdf_logexp(self, x, **params):
         return np.exp(self._logpdf_dispatch(x, **params))
 
+    @_set_invalid_nan
+    def logpmf(self, x, /, *, method=None):
+        return self._logpmf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _logpmf_dispatch(self, x, *, method=None, **params):
+        if self._overrides('_logpmf_formula'):
+            method = self._logpmf_formula
+        elif _isnull(self.tol):  # ensure that developers override _logpmf
```

Comment:
Just mirroring `_pdf_dispatch` logic here. This was always a placeholder for better logic. The concern is that the PMF can underflow before the log is taken, and the hope is that annoying failures here will encourage distributions authors to override `_logpmf_formula`.